### ISSUE TITLE
[UI Change Request] Route calculation progress bar

### ISSUE THREAD

**Issue by franco999** [osmandapp/Osmand#4149]
What do you think of changing the progress bar when calculating the route? Something similar to the screenshot.

That is, instead of placing it at the top of the screen and with a very small size, place it just above the menu and occupying the entire width of the screen.

![r](https://user-images.githubusercontent.com/12088580/28395312-7192f640-6cc9-11e7-8611-33229ce5907b.jpg)

**Comment by sonora** [comment]
Personally, I like that!

**Comment by polarbearing** [comment]
+1, less coverage of the map, and it would solve #3010.

In this context, it would be good to indicate in which of the phases the calculation is, e.g. by changing the colour of the bar, or growing left->right for the first and right->left in the second phase.

**Comment by vshcherb** [comment]
What to do if route recalculation happens without that dialog.

**Comment by franco999** [comment]
@vshcherb In any case, could you leave both progress bars, until you find a better solution?

I proposed it because sometimes when the route is calculated, the current bar is not seen and you have to exit the menu and zoom in to make it appear.

**Comment by sonora** [comment]
When the dialog is not visible, we can put it to the very bottom of the screen. Or alternatively, we can always put the progress indicator to the very top edge of the screen, as browsers like Firefox do it for loading pages, or some web sites like github conventionally do it..

**Comment by franco999** [comment]
@sonora I was thinking the same. I do not know if it will be difficult or not to program it, but it is a good idea.

**Comment by ignaciolep** [comment]
Great idea!

**Comment by sonora** [comment]
@vshcherb I guess this is another candidate to still include in the next release?

**Comment by vshcherb** [comment]
Oh yeah... somehow we missed that small improvement mainly because it will be a duplicate and not replacement cause we still need a topbar for recalculation at least

**Comment by sonora** [comment]
aA I said above: We can always put the progress indicator to the very top edge of the screen, as browsers like Firefox do it for loading pages. In this fashion we could truly replace the current indicator.

**Comment by AlexSyt** [comment]
I added a progress bar above the menu.

**Comment by sonora** [comment]
Yeah, looking good, I think! Now we can move the old progress bar to span the very top of the screen, and hide it as long as the new one on the dialog is visible.

**Comment by AlexSyt** [comment]
@sonora , I will move the old progress bar to span the very bottom of the screen :)

**Comment by AlexSyt** [comment]
@franco999 , You are welcome)

**Comment by AlexSyt** [comment]
@sonora , @franco999 , progress bar moved to the very bottom of the screen, take a look, please)

**Comment by franco999** [comment]
@AlexSyt 
1.  Remove empty white square
2.  In my opinion (I don't know the design standard), I think it would look better on top.  At the bottom, it gives me the feeling that it's a little hidden.

But in general, I like him.

**Comment by sonora** [comment]
Alexander, great work, thank you so much!

Maybe the screen bottom is not even so bad, because it is where the dialog slides down to and then only the bar remains there, so probably just a matter of getting used to. I like it!

One more small bug I find in addition: If during a route calculation you first hide the route dialog and then tap an empty (no POIs or tappable objects) area on the map (this hides all map screen widgets), the old (top center) progress bar appears again. But I guess we should also here simply show the new (bottom) bar, right?

**Comment by AlexSyt** [comment]
@franco999 
1. I don`t have that white square, but I`ll try to figure it out.
2. Progress bar at the top is complicated because of different OsmAnd`s top-bars.
3. I`ll hide the bottom progress in that situation.

**Comment by AlexSyt** [comment]
@sonora , that progress bar is needed for the "Waypoints" screen, maybe we`ll do something about it later.

**Comment by sonora** [comment]
Yes fine, I very much like what we have achieved so far. Thanks again!

**Comment by franco999** [comment]
1.  I'll try to reinstall it completely, maybe it'll work like this.
2.  With what @sonora said earlier, I like it on the bottom. 😅

**Comment by AlexSyt** [comment]
@franco999 , just fixed 1) and 3)

**Comment by franco999** [comment]
4. Is it also possible to modify the bar in the "Configure screen" and "Configure map" menus?
e.g. 
![f](https://user-images.githubusercontent.com/12088580/40550855-a31b0702-6011-11e8-9e24-50300531a3e0.jpg)

**Comment by AlexSyt** [comment]
It is complicated, like in the "Waypoints" screen too. Maybe we'll do it later.

**Comment by franco999** [comment]
No problem. 👍

**Comment by sonora** [comment]
I have a few minor observations we may want to look at at some point, as follows:
- **Color:** The progress bar now borders on the map, so it should have a color with good contrast vs. the map. I suggest instead of the generic light blue we could use the route color as defined by the rendering style? (Which is an intense dark blue for most cases)
- **Placement:** On some devices (maybe those with non-standard screen sizes, not sure) like the Samsung S4 Mini, the progress bar when attached to the route config dialog is not at the very top of the dialog but about 1.5 progress bar widths further down. Conversely, when the dialog is hidden, the progress bar is not at the very bottom of the map screen, but about 2 progress bar widths up from the bottom (with a small strip of the map being visible between the bar itself and the lower screen edge). Probably a rounding error of some sort when placing the bar?
- **Value:** We have always had this: when our normal 2 phase routing is active, the progress bar runs across the screen twice before the route is finished. Should not be a problem to predict when 2 phases will occur and simply adjust so that the end of the first phase is simply the 50% value (screen center), and the right hand screen edge is only reached after the second round (100% value)?
- **Initially no progress visible** for some time after route calculation starts: I have seen users stare at the route config dialog wondering if they need to still tap anything because the route calculation had not visibly started yet. (This e.g. can be due to the position still not found so the app waiting for that, or some initialization still going on, see e.g. #4043). This could maybe be solved by immediately starting the progress bar at its say 5% level (if a destination is already selected, not to be confused with if not, i.e. where the user really has to become active first), so the user clearly sees if OsmAnd is in the progress of calculating already, or if a user input is still needed?

**Comment by sonora** [comment]
PS: Regarding "placement": Here is what I mean. And the issue may actually be connected to the fact that on those devices also the start/end point text like "My Position" seems to be placed too low, as well as the text in all map widgets like the speed, distance to destination, etc. Not sure what this is, something about how placement and margins are treated. But I see this on some devices only, S4 Mini being an example.

![untitled](https://user-images.githubusercontent.com/845735/40576918-0cca78ee-60fe-11e8-9b26-e17642209274.png)

**Comment by scaidermern** [comment]
As already noted by @sonora, the current solution is hardly readable because the bar is too thin. While driving it is hard to know if OsmAnd is "idling" or (re-)calculating a route. Please increase the progress bar height.

**Comment by vshcherb** [comment]
Linked to #5506
